### ISSUE TITLE
Does anyone account this error(content-length mismatch)?

### ISSUE THREAD

**Issue by ReginaLiang** [awslabs/aws-lambda-web-adapter#552]
I used the nextjs, then fetch api, will occur `Request body length does not match content-length header`. 
How to fix it?

Thanks.

**Comment by bnusunny** [comment]
@ReginaLiang Could you provide more details about how this error happened? The runtime version, the LWA version, the invoke mode, and how frequent did it happen?

**Comment by ReginaLiang** [comment]
> @ReginaLiang Could you provide more details about how this error happened? The runtime version, the LWA version, the invoke mode, and how frequent did it happen?

Sure, I used lambda to deploy a nextjs app, and the runtime version is Nodejs 18, the LWA version is 0.8.4. Then there is an api gateway http api to call this lambda. HTTP API sets up a proxy for all requests. Finally, Nextjs application can run successfully, But when fetching the background API, it will report `Request body length does not match content-length header`. This problem only occurs in get requests, Post requests are fine.

**Comment by bnusunny** [comment]
@ReginaLiang can you share more CloudWatch Logs for the error message?

**Comment by ReginaLiang** [comment]
> @ReginaLiang can you provide more CloudWatch Logs related to the error message?

<img width="1256" alt="截屏2024-11-29 12 28 43" src="https://github.com/user-attachments/assets/f54b29f6-89d7-45af-9aca-f3f30704c507" />

FYI, This error is from Lambda. Thanks.

**Comment by bnusunny** [comment]
From these logs, it seems the next.js app was trying to send a GET request to another API and failed with the error `Request body length does not match content-length header`.  Could you log the actual body lenthg and the content-length header before sending the request? Does this work on you local computer?

**Comment by ReginaLiang** [comment]
> From these logs, it seems the next.js app was trying to send a GET request to another API and failed with the error `Request body length does not match content-length header`. Could you log the actual body lenthg and the content-length header before sending the request? Does this work on you local computer?

Yes, It works fine on my local computer. and the body as follows:
<img width="953" alt="截屏2024-11-29 11 54 22" src="https://github.com/user-attachments/assets/27a32fe7-0e37-4f4d-a1ad-9528241131ec" />

**Comment by bnusunny** [comment]
This is the Lambda payload. LWA will translate it into a HTTP request and send to your next.js web app. Could you add debug logging in your next.js web app to show the http request it received? Especially the content-length header and the actual body.

**Comment by ReginaLiang** [comment]
> This is the Lambda payload. LWA will translate it into a HTTP request and send to your next.js web app. Could you add debug logging in your next.js web app to show the http request it received? Especially the content-length header and the actual body.

In fact, my nextjs app does not receive any parameters from lambda, but simply requests other apis through fetch, Will this have an impact?

**Comment by bnusunny** [comment]
Hmm, this doesn't seem like an issue with LWA. Could you share a simple example which can reproduce this issue?

**Comment by ReginaLiang** [comment]
> Hmm, this doesn't seem like an issue with LWA. Could you share a simple example which can reproduce this issue?

I will prepare a sample to reproduce this problem in my free time. I appreciate your help.

**Comment by bnusunny** [comment]
I will close this issue for now. Reopen if you find this is related to LWA.
